### DISCUSSION TITLE
Auto format range

### DISCUSSION THREAD

**Discussion by noahfraiture** [LazyVim/LazyVim#2228] [Q&A]
Hey, I searched for a while I didn't find the answer, maybe because I don't have the rights words. Auto formatting is great but I find the break line too short and I would like to increase its size to have longer line. Do you know where I can find that ? I didn't find anything in the format except in [https://github.com/stevearc/conform.nvim](conform github page) but I don't understand how to edit it. Here's the code I saw `      -- If the formatter supports range formatting, create the range arguments here
      range_args = function(ctx)
        return { "--line-start", ctx.range.start[1], "--line-end", ctx.range["end"][1] }
      end,`

Thank you for any help !

**Comment by dpetka2001** [comment]
You should configure that as an argument for the formatter you use. For example, for `stylua`, you would add `prepend_args = { "--column-width", "50" },`. But that option may not be the same for another formatter. For `black` (Python formatter) the option would be `--line-length`. 
So, it depends. You should use the `help` or docs for your formatter to see how you can achieve it.

By the way, I think you should really avoid to integrate that directly in `conform`, because different projects might use different
style policies, so it'd be best to just configure this in the configuration file of the formatter you use at the root of your corresponding
project.

**Reply by noahfraiture** [reply]
Oh ok so it's language formatter dependent, sad
Thank you !

**Comment by noahfraiture** [comment]
I reopen this discussion as I don't succeed. I tried to make this number greater and even in stylua.toml but nothing change, black still want to break line for no reason. Do you have any idea ? I didn't find the solution in options.lua either

**Reply by dpetka2001** [reply]
Please read the official black [docs](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-via-a-file) for how to configure it. This doesn't have anything to do with stylua or options.lua either.
Each language has different formatters and each formatter is configured via a configuration file. You should read the docs of the formatter for which configuration file to use.
